Configure code formatter for Java code

nastra opened the pull request:
**PR Checklist**

- [x] A description of the changes is added to the description of this PR.
- [ ] If there is a related issue, make sure it is linked to this PR.
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added or modified a feature, documentation in `docs` is updated

**Description of changes**

This configures and applies spotless for Java code. The PR is currently WIP and requires some additional documentation so that it's easier for devs to get started and include `bin/sbt spotlessApply` into their workflow when Java code formatting is off.

creechy commented:
+1 for adding and enforcing code formatting. You don't mention the code formatting standard, but I assume its Google Java Format. Just going to throw it out there to maybe investigate Palantir Java Format. IMO Google Java Format has issues, particularly with closures/lambdas. I don't know a lot about Palantir Java Format but you can see some comparisons between the two here - https://github.com/palantir/palantir-java-format - and it seems like Spotless supports it too.

tdas commented:
@nastra some initial feedback
1. Its really hard to review this much amount of code difference. as I suggested in the issue, is it possible to break it down into smaller PRs. 
2. we should not modify client as its generated. so spotless should ignore client

tdas commented on .github/workflows/spotless.yml:
```diff
@@ -0,0 +1,27 @@
+# separate terms of service, privacy policy, and support
+# documentation.
+
+name: Check code style
```
I am trying to standardize these names to be nouns. so I suggest "Code style check"

nastra commented:
> @nastra some initial feedback
> 
> 1. Its really hard to review this much amount of code difference. as I suggested in the issue, is it possible to break it down into smaller PRs.
> 2. we should not modify client as its generated. so spotless should ignore client. Similar models in the server should be ignored as well as they are generated.

I included the formatting changes initially as an experimentation to give people an overview of how things would look, but I agree that this is difficult to review, so maybe as a first step we should only add the respective configuration and later apply the formatting.

@creechy just FYI I had to switch from **sbt-spotless** to https://github.com/sbt/sbt-java-formatter due to a bug in **sbt-spotless** (it's unfortunately externally maintained and the last release was in 2020). **sbt-java-formatter** produces the exact same formatting results with the one tradeoff that you can't switch the formatter as it uses **google-java-format**.

nastra commented:
@tdas I've updated the PR to only include the the respective configuration for auto code formatting. Code can be automatically formatted by running `build/sbt javafmtAll` (the `All` includes the `Test` configuration). 
That code style is currently not enforced, which we can later change.
I've added a note on how to configure IntelliJ so that code formatting in the IDE and on the cmd line produces the same results.

tdas commented on README.md:
````diff
@@ -153,6 +153,17 @@ See the full [tutorial](docs/tutorial.md) for more details.
   ``` 
   This will regenerate the OpenAPI data models in the UC server and data models + APIs in the client SDK.
 
+## Setting up IDE and Code Style
````
```suggestion
## Setting up IDE
```

tdas commented:
This PR looks very good! Lets just fix the conflicts, and then I will merge this. thank you @nastra for do this!

nastra commented:
Thanks @tdas, I've resolved the conflicts and pushed a new version

tdas commented on README.md:
```diff
@@ -163,6 +163,19 @@ See the full [tutorial](docs/tutorial.md) for more details.
   This will regenerate the OpenAPI data models in the UC server and data models + APIs in the client SDK.
+
+
+## Setting up IDE
```
I will merge this PR but I think this section needs basic instructions on how to set up IntelliJ / Eclipse. Code formatting is the not the only thing in the IDE set up.. is it?

tdas replied:
can you improve it in a follow up PR?

nastra replied:
@tdas just to confirm: Installing the `google-java-format` plugin is really the only step required and https://github.com/google/google-java-format#intellij-android-studio-and-other-jetbrains-ides contains all the steps required to install + enable it. I figured it should be fine to refer to the original docs for this rather than try and duplicate something here which might eventually get outdated.
Let me know if you still think that we should add some more text here

tdas replied:
The codestyle part is fine. But since we are making a general section on "Setting up IDE", I think its good to say how to open up the project in IntelliJ or Eclipse. Just a simple thing like load the build.sbt (at least for IntelliJ that I use) is sufficient.

nastra replied:
I've opened https://github.com/unitycatalog/unitycatalog/pull/170 to address that
